paimon-core/src/main/java/org/apache/paimon/operation/AbstractFileStoreWrite.java

```diff
@@ -272,6 +272,13 @@ public void restore(List<State> states) {
                             state.commitIncrement,
                             compactExecutor());
             notifyNewWriter(writer);
+
+            try {
```

Comment:
revert here, restore is just for schema change, we don't need to create a compaction.